src/settings.js

```diff
@@ -1630,6 +1630,16 @@ var ABORT_ON_WASM_EXCEPTIONS = 0;
 // Implies STANDALONE_WASM.
 var PURE_WASI = 0;
 
+// Set to 1 to define the WebAssembly.Memory object outside of the wasm
+// module.  By default the wasm module defines the memory and exports
+// it to JavaScript.
+// Use of the following settings will enable this settings since they
+// depend on being able to define the memory in JavaScript:
+// - USE_PTHREADS
+// - RELOCATABLE
+// - ASYNCIFY_LAZY_LOAD_CODE
+var EXTERNAL_MEMORY = 0;
```

Comment:
How about `IMPORTED_MEMORY`?

Reply:
I'm trying to avoid the terms "import" and "export" here because that can be ambiguous depending on if we are talking about the JS or the Wasm side of the fence.

The term "external" feel less ambiguous to me because it implies the memory is *not* defined in wasm.   I'm attempting to come up with a phrase that means something like:

- `MEMORY_DEFINED_OUTSIDE_OF_WASM`
- `MEMORY_DEFINED_IN_JS`
- `MEMORY_DEFINED_AT_RUNTIME_BY_USER`

When trying come up with an adjective for this ,`external` seemed more meaningful than `imported`.

Reply:
I don't agree it's ambiguous. We have wasm imports and exports. We do not talk about a wasm export as an "import from wasm to JS",  that I can think of? We call it a wasm export.

I'd say that "external" is more ambiguous - external to what? The wasm, or maybe the entire JS?

Reply:
My idea was that `external` was supposed to have that wider meaning yes, since it implies that the user can provide there own memory completely externally if they want to.

I will will rename.  Hopefully very few folks should need to use this option.